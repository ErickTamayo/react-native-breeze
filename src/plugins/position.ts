import { PluginFunction } from "./types";

export type PluginGroups = {
  position: "absolute" | "relative";
};

export const pattern = /^(?<position>absolute|relative)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { position } = groups;
  return { position };
};
