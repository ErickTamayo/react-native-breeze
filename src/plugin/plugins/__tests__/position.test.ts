import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
  shouldParseCorrectlyTest,
} from "../../utils/tests";
import { pattern, plugin } from "../position";

describe("position", () => {
  const shouldMatch = ["absolute", "relative"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
