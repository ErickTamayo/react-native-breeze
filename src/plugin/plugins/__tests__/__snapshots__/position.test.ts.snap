// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`position Should parse (absolute) correctly 1`] = `
Object {
  "position": "absolute",
}
`;

exports[`position Should parse (relative) correctly 1`] = `
Object {
  "position": "relative",
}
`;
